src/services/SelfId/SelfId.ts

```diff
@@ -0,0 +1,73 @@
+import { Address } from 'types'
+import { getRandomDefaultImage } from 'utils/identicon'
+import { Core } from '@self.id/core'
+import { getLegacy3BoxProfileAsBasicProfile } from '@self.id/3box-legacy'
+import { IdxUser } from './types'
+
+const core = new Core({ ceramic: 'https://gateway.ceramic.network' })
+
+const ipfsGateway = 'https://ipfs.infura.io/ipfs/'
+
+type User = {
+  status?: string
+  name?: string
+  image: string
+}
+
+type UserWithCache = User & {
+  noCache?: boolean
+}
+
+const transformIdxUser = (user: IdxUser): User => {
+  const imageSrc = user?.image?.original?.src
+  let image = ''
+  if (imageSrc && imageSrc.startsWith('ipfs://')) {
+    const cid = imageSrc.substring(7)
+    image = `${ipfsGateway}${cid}`
+  }
+  return {
+    image
+  }
+}
+
+export const getSelfIdProfile = async (
+  wallet: Address
+): Promise<UserWithCache> => {
+  let profile: User | null = null
+
+  try {
+    let idxUser: IdxUser | null = await core.get(
```

Comment:
Nonblocking, but we should look into the multi get api which apparently exists, but doesn't seem to be documented

Reply:
Also not blocking, but it looks like their api allows for fetching images of set dimensions - https://developers.ceramic.network/reference/self-id/modules/image_utils/#get-the-most-suitable-image-source-for-given-dimensions
which would be nice to add eventually

Reply:
Reference for how to call getMultiple (or at least my best guess, digging into the code for a couple min)

Covert all ids to did first, ref:
https://github.com/ceramicstudio/self.id/blob/34f04f8203eff8c093193d6012ab0796289fb2c8/packages/core/src/core.ts#L135-L141

Get the Did Store: https://github.com/ceramicstudio/self.id/blob/34f04f8203eff8c093193d6012ab0796289fb2c8/packages/core/src/core.ts#L90-L93

Call Get Multiple:
https://github.com/ceramicstudio/js-glaze/blob/a1897bb354a3e511ea1bf4cd526c3105ff108478/packages/did-datastore/src/index.ts#L326-L363

Also if this works, maybe worth a PR into the `self.id/core` package